### ISSUE TITLE
Support for Explicitly Declaring Implementation of Protocols according to PEP 544

### ISSUE THREAD

**Issue by lambdakris** [microsoft/pyright#4199]
**Is your feature request related to a problem? Please describe.**
I was reading through the [PEP 544](https://peps.python.org/pep-0544/#explicitly-declaring-implementation) and noticed [the section that discusses "Explicitly declaring implementation"](https://peps.python.org/pep-0544/#explicitly-declaring-implementation) seems to suggest that when a class explicitly declares that it implements a protocol, a type checker could check whether the class properly implements it (which I assume means it checks whether the class implements the members of the protocol). I tried that in Pyright using strict mode but I did not get any errors or warnings and I was hopping that there was a way to enable this as this would be a very excellent tool in the toolbelt for enabling earlier type safety. I know that this is technically not "duck typing", but it seems like [PEP 544 does allow for this](https://peps.python.org/pep-0544/#explicitly-declaring-implementation), so I figured I would ask. Below is an example copied from the [Explicitly declaring implementation section of PEP 544](https://peps.python.org/pep-0544/#explicitly-declaring-implementation).

```python
class RGB(Protocol):
    rgb: Tuple[int, int, int]

    @abstractmethod
    def intensity(self) -> int:
        return 0

class Point(RGB):
    def __init__(self, red: int, green: int, blue: int) -> None:
        self.rgb = red, green, blue  # Error, 'blue' must be 'int'
        
# Type checker might warn that 'intensity' is not defined
```

**Describe the solution you'd like**

I would like it if there was a setting that I could turn on in Pyright that would warn or error when a class declares that it implements a Protocol but it does not declare the members that the protocol declares. I am uncertain whether this should be confined to only Protocol members that have the `@abstractmethod` attribute or whether it is sufficient to just have the convention that if this Pyright setting is turned on, then a class is expected to declare all members of a Protocol, whether annotated as abstract or not. I figure I will leave that question to folks who are more familiar with language and compiler design :)!

**Additional context**

**Comment by erictraut** [comment]
`@abstractmethod` is meant to be used within a subclass of `ABC`, not a `Protocol`. The `@abstractmethod` decorator has no meaning in a `Protocol`.

The standard way to indicate an unimplemented method within a protocol is to use `...` in its body. If you supply other statements like `return 0`, then pyright assumes that it is implemented.

If you change your definition of `RGB` to the following, you'll receive the errors you expect.

```python
class RGB(Protocol):
    rgb: tuple[int, int, int]

    def intensity(self) -> int:
        ...
```
